### DISCUSSION TITLE
Write xarray zarr to Blob storage?

### DISCUSSION THREAD

**Discussion by elliot42** [microsoft/PlanetaryComputer#127] [Q&A]
Sorry if I'm missing something, is there a way to write xarray zarrs back to Blob storage?

The docs have great examples of _reading_ xarray/zarrs from storage (which works great), but I haven't been able to find writes, e.g. here:

https://planetarycomputer.microsoft.com/docs/quickstarts/storage/

My understanding is that using the `dataset.to_zarr()` interface should be particularly great in this context, because it can chunk and parallelize the writes back to the cloud storage, in cloud storage optimized format, for datasets larger than Hub/VM RAM--but I'm getting exceptions trying to write to either `https://...blob.core.windows` or `abfs://` URLs.  It _seems_ like these optimizations are little more straightforward via the `.to_zarr()` interface rather than manual `blob_client.upload_blob()` example, but I could totally be missing stuff.

Any thoughts greatly appreciated, thank you!

**Comment by TomAugspurger** [comment]
Good point, that is missing from our documentation. https://planetarycomputer.microsoft.com/docs/quickstarts/storage/ shows writing a COG to blob storage, but I'll update that to show Zarr as well.

It should be something like

```python
store = adlfs.AzureBlobFilesystem(account_name, credential=sas_token).get_mapper("<container-name>/path")
ds.to_zarr(store)
```

**Comment by elliot42** [comment]
This works!!  Minor capitalization detail in `adlfs.AzureBlobFileSystem`.

And then can re-read the written results via `xarray.open_dataset(store, engine='zarr')`

So awesome, thanks so much!  I'm happy to help with docs PRs if there's a public repo for them.  Thanks again, platform is super exciting!
